app/src/main/java/space/taran/arkmemo/data/repositories/TextNotesRepository.kt

```diff
@@ -17,69 +20,156 @@ import kotlin.io.path.getLastModifiedTime
-                if (filePath.fileName.extension == NOTE_EXT) {
-                    number += 1
+            val versionDir = File(path.toFile(), ARK_VERSIONS_DIR)
+            if (versionDir.exists() && versionDir.isDirectory) {
+                Files.list(versionDir.toPath()).forEach { filePath ->
                     try {
-                        val jsonFile = filePath.toFile()
-                        val fileReader = FileReader(jsonFile)
-                        val bufferedReader = BufferedReader(fileReader)
-                        val jsonTextNote = StringBuilder()
-                        with(bufferedReader) {
-                            forEachLine {
-                                jsonTextNote.append(it)
-                            }
-                            val content = JsonParser.parseNoteFromJson(jsonTextNote.toString())
-                            val size = Files.size(filePath)
-                            val id = computeId(size, filePath)
-                            val meta = ResourceMeta(
-                                id,
-                                filePath.fileName.toString(),
-                                filePath.extension,
-                                filePath.getLastModifiedTime(),
-                                size
-                            )
-
-                            val note = TextNote(content, meta)
-                            notes.add(note)
-                            close()
-                        }
+                        val fileName = filePath.fileName.toString()
+                        val rootResourceId: ResourceId = fileName.toLong()
+                        val meta = VersionMeta(
+                            rootResourceId
+                        )
+                        val versionStorage = VersionStorage(getPath()!!)
+                        val verContent = Version.Content(versionStorage.versions(rootResourceId))
+                        val ver = Version(verContent, meta)
+                        versions.add(ver)
                     } catch (e: Exception) {
                         e.printStackTrace()
                     }
                 }
             }
         }
-        return notes
+        return versions
     }
 
-    private fun createTextNoteFile(path: Path?, noteString: String?) {
+    private fun getLastChildrenId(rootResourceId: ResourceId): ResourceId {
+        val path = getPath()
+        var lastChildResourceId: ResourceId? = null
+        if (path != null) {
+            val versionDir = File(path.toFile(), ARK_VERSIONS_DIR)
+            val versionFile = File(versionDir, rootResourceId.toString())
+            if (versionFile.exists()) {
+                val fileReader = FileReader(versionFile)
+                val bufferedReader = BufferedReader(fileReader)
+                val jsonVersion = StringBuilder()
+                with(bufferedReader) {
+                    forEachLine {
```

Comment:
But can we have more than 1 line in the version file?
Am I right, the format is like this?
```
rootId, version1, version2 ... latestVersion
```

Reply:
Or if it's JSON, then something like this?
```
{ root: ResourceId, children: Array<ResourceId> }
```

Reply:
Its JSON, right now the format its that the rootId its the filename, and the versions are contained inside the file:
`{ idList: [] }`
 I think we have talked about putting the rootId as the first item in the list, so i will stop using VersionMeta, then the rootId will just be the first item, and the filename also.
 We have talked about that we should only write to file in version when there is more than one version, so i will apply this too, so then the notes from main branch will be compatible.